From 0b67fb1538fb411608763c2133f642643801d121 Mon Sep 17 00:00:00 2001
From: Xinxiang Wang <wxinxiang8@gmail.com>
Date: Thu, 13 Aug 2026 13:32:19 +0800
Subject: [PATCH 38/44] dt-bindings: media: i2c: add OmniVision OV13850 binding

Add Device Tree binding document for the OmniVision OV13850,
a 13-megapixel (4208x3120) CMOS image sensor controlled via
an I2C-compatible SCCB bus.

The sensor transmits images over MIPI CSI-2 with 1 or 2 data
lanes. The binding describes the required clock input (xvclk),
optional reset and powerdown GPIOs, and the CSI-2 transmitter
port.

Signed-off-by: Xinxiang Wang <wxinxiang8@gmail.com>
---
 .../bindings/media/i2c/ovti,ov13850.yaml      | 103 ++++++++++++++++++
 1 file changed, 103 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/media/i2c/ovti,ov13850.yaml

diff --git a/Documentation/devicetree/bindings/media/i2c/ovti,ov13850.yaml b/Documentation/devicetree/bindings/media/i2c/ovti,ov13850.yaml
new file mode 100644
index 0000000000000000000000000000000000000000..f18a693a3c5960d5746f4ebae63046d7280a001d
--- /dev/null
+++ b/Documentation/devicetree/bindings/media/i2c/ovti,ov13850.yaml
@@ -0,0 +1,103 @@
+# SPDX-License-Identifier: (GPL-2.0 OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/media/i2c/ovti,ov13850.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: OmniVision OV13850 Image Sensor
+
+maintainers:
+  - Xinxiang Wang <wxinxiang8@gmail.com>
+
+description: |
+  The OmniVision OV13850 is a color CMOS 13 Megapixels (4208x3120) image
+  sensor controlled through an I2C-compatible SCCB bus. The sensor transmits
+  images on a MIPI CSI-2 output interface with up to 4 data lanes.
+
+allOf:
+  - $ref: /schemas/media/video-interface-devices.yaml#
+
+properties:
+  compatible:
+    const: ovti,ov13850
+
+  reg:
+    maxItems: 1
+
+  clocks:
+    description: XVCLK external clock
+    maxItems: 1
+
+  clock-names:
+    const: xvclk
+
+  reset-gpios:
+    description: XSHUTDN reset GPIO (active HIGH)
+    maxItems: 1
+
+  powerdown-gpios:
+    description: PWDNB powerdown GPIO (active HIGH)
+    maxItems: 1
+
+  port:
+    $ref: /schemas/graph.yaml#/$defs/port-base
+    additionalProperties: false
+    description: MIPI CSI-2 transmitter port
+
+    properties:
+      endpoint:
+        $ref: /schemas/media/video-interfaces.yaml#
+        unevaluatedProperties: false
+
+        properties:
+          data-lanes:
+            oneOf:
+              - items:
+                  - const: 1
+              - items:
+                  - const: 1
+                  - const: 2
+          link-frequencies: true
+
+        required:
+          - data-lanes
+          - remote-endpoint
+
+required:
+  - compatible
+  - reg
+  - clocks
+  - port
+
+unevaluatedProperties: false
+
+examples:
+  - |
+    #include <dt-bindings/gpio/gpio.h>
+    #include <dt-bindings/pinctrl/rockchip.h>
+
+    i2c {
+        #address-cells = <1>;
+        #size-cells = <0>;
+        status = "okay";
+
+        ov13850: camera@10 {
+            compatible = "ovti,ov13850";
+            reg = <0x10>;
+
+            clocks = <&xvclk>;
+            clock-names = "xvclk";
+
+            reset-gpios = <&gpio3 RK_PA7 GPIO_ACTIVE_HIGH>;
+            powerdown-gpios = <&gpio3 RK_PC5 GPIO_ACTIVE_HIGH>;
+
+            port {
+                ov13850_out: endpoint {
+                    remote-endpoint = <&mipi_in_cam0>;
+                    data-lanes = <1 2>;
+                };
+            };
+        };
+    };
+
+...
-- 
2.53.0

